docs/source/distributed.tensor.parallel.rst

```diff
@@ -1,7 +1,59 @@
 .. role:: hidden
     :class: hidden-section
 
-Tensor Parallelism
-========================
-.. py:module:: torch.distributed.tensor.parallel
+Tensor Parallelism - torch.distributed.tensor.parallel
+======================================================
+
+We built Tensor Parallelism(TP) on top of DistributedTensor(DTensor) and
+provide several Parallelis styles: Rowwise, Colwise and Pairwise Parallelism.
+
+The entrypoint to parallelize your module and use tensor parallelism is:
+
+.. automodule:: torch.distributed.tensor.parallel
+
 .. currentmodule:: torch.distributed.tensor.parallel
+
+.. autofunction::  parallelize_module
+
+All parallel styles inherits the following:
+
+.. autoclass:: torch.distributed.tensor.parallel.style.ParallelStyle
+  :members:
+
+We have following parallel styles:
+
+.. autoclass:: torch.distributed.tensor.parallel.style.RowwiseParallel
+  :members:
+
+.. autoclass:: torch.distributed.tensor.parallel.style.ColwiseParallel
+  :members:
+
+.. autoclass:: torch.distributed.tensor.parallel.style.PairwiseParallel
+  :members:
+
+Because we use DTensors within the TP module, we need to specify how we want
+it to interact with the module before and after. The followings are functions
+used for input/output preparation:
+
+
+.. currentmodule:: torch.distributed.tensor.parallel.style
+
+.. autofunction::  make_input_replicate_1d
```

Comment:
Actually I'm wondering if we make those APIs be something like `mark_input_replicate_1d` instead of `make`? The real thing we are doing is to mark the inputs/outputs instead of turning them into (i.e. we are using `from_local` instead of `distribute_tensor`)

Reply:
I understand where you are from... Although internally we don't have drastical changes here. From users' perspective we still change a Tensor to DTensor. So I really don't like the word "mark". (Definition: https://www.merriam-webster.com/dictionary/mark). It does not contain any meaning related to change. I would prefer verbs like "change", "convert", "transform", etc.. Or even construct maybe..